### ISSUE TITLE
Browser View add-on fails to build with aarch64

### ISSUE THREAD

**Issue by decuple-dream** [zaproxy/zaproxy#7707]
### Describe the bug

Hi, Teams! 👋🏼 how are you? 
There's a problem. An error occurs while building by referring to the official build documentation(https://www.zaproxy.org/docs/developer/quick-start-build/) of zaproxy.

### Steps to reproduce the behavior

First, place the zaproxy and zap-extension folders in the same root directory.
And go to the zap-extension folder and run `./gradlew copyZapAddOn`.

./gradlew copyZapAddOn
```
> Configure project :addOns:browserView
Project :addOns:browserView => no module-info.java found

> Task :addOns:browserView:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':addOns:browserView:compileJava'.
> Could not resolve all files for configuration ':addOns:browserView:compileClasspath'.
   > Could not find javafx-web-11-mac-aarch64.jar (org.openjfx:javafx-web:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-web/11/javafx-web-11-mac-aarch64.jar
   > Could not find javafx-controls-11-mac-aarch64.jar (org.openjfx:javafx-controls:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-controls/11/javafx-controls-11-mac-aarch64.jar
   > Could not find javafx-media-11-mac-aarch64.jar (org.openjfx:javafx-media:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-media/11/javafx-media-11-mac-aarch64.jar
   > Could not find javafx-swing-11-mac-aarch64.jar (org.openjfx:javafx-swing:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-swing/11/javafx-swing-11-mac-aarch64.jar
   > Could not find javafx-graphics-11-mac-aarch64.jar (org.openjfx:javafx-graphics:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-graphics/11/javafx-graphics-11-mac-aarch64.jar
   > Could not find javafx-base-11-mac-aarch64.jar (org.openjfx:javafx-base:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/11/javafx-base-11-mac-aarch64.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
```

Next, go to zap-hud and run `./gradlew copyZapAddOn`.

```
./gradlew copyZapAddOn

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nodeSetup'.
> Failed to query the value of task ':nodeSetup' property 'nodeArchiveFile'.
   > Could not resolve all files for configuration ':detachedConfiguration1'.
      > Could not find org.nodejs:node:10.24.1.
        Searched in the following locations:
          - https://repo.maven.apache.org/maven2/org/nodejs/node/10.24.1/node-10.24.1.pom
          - https://nodejs.org/dist/v10.24.1/node-v10.24.1-darwin-arm64.tar.gz
        Required by:
            project :

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
10 actionable tasks: 10 up-to-date
```

### Expected behavior

According to the official documentation, it should build without any errors.
But distMacArm64 is well build.

./gradlew distMacArm64
```
> Task :zap:downloadMacOsJreArm64
Download https://api.adoptium.net/v3/binary/version/jdk-11.0.17+8/mac/aarch64/jre/hotspot/normal/eclipse?project=jdk

> Task :zap:distMacArm64
created: /Users/Eo/zaproxy/zap/build/distributions/ZAP_2.13.0-SNAPSHOT_aarch64.dmg
```
Can I just use it like this?

## Gradle Version
* Build Time: 2022-11-25 13:35:10 UTC
* Revision: daece9dbc5b79370cc8e4fd6fe4b2cd400e150a8
* Kotlin: 1.7.10
* Groovy: 3.0.13
* Ant: Apache Ant(TM) version 1.10.11 compiled on July 10 2021
* JVM: 17.0.6 (Homebrew 17.0.6+0)
* OS: Mac OS X 13.1 aarch64

### Would you like to help fix this issue?

- [ ] Yes

**Comment by thc202** [comment]
The problem is the architecture not the Gradle version (wrapper uses 7.5.x).

**Comment by decuple-dream** [comment]
When will it be patched? Even though I have a new github, the error occurs the same.
```
./gradlew copyZapAddOn

> Configure project :addOns:browserView
Project :addOns:browserView => no module-info.java found

> Task :addOns:browserView:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':addOns:browserView:compileJava'.
> Could not resolve all files for configuration ':addOns:browserView:compileClasspath'.
   > Could not find javafx-web-11-mac-aarch64.jar (org.openjfx:javafx-web:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-web/11/javafx-web-11-mac-aarch64.jar
   > Could not find javafx-controls-11-mac-aarch64.jar (org.openjfx:javafx-controls:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-controls/11/javafx-controls-11-mac-aarch64.jar
   > Could not find javafx-media-11-mac-aarch64.jar (org.openjfx:javafx-media:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-media/11/javafx-media-11-mac-aarch64.jar
   > Could not find javafx-swing-11-mac-aarch64.jar (org.openjfx:javafx-swing:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-swing/11/javafx-swing-11-mac-aarch64.jar
   > Could not find javafx-graphics-11-mac-aarch64.jar (org.openjfx:javafx-graphics:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-graphics/11/javafx-graphics-11-mac-aarch64.jar
   > Could not find javafx-base-11-mac-aarch64.jar (org.openjfx:javafx-base:11).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/11/javafx-base-11-mac-aarch64.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
239 actionable tasks: 4 executed, 235 up-to-date
```
| zap-extension

**Comment by thc202** [comment]
Whenever is deemed good, if this is blocking you you can apply the (potential) fix locally and provide feedback.

**Comment by decuple-dream** [comment]
thanks to reply @thc202!, I not understand your mean. 
Does the build go without errors on the current aarch64 version?

**Comment by thc202** [comment]
Hopefully it should in the referenced pull request, you just need to change this line locally: https://github.com/zaproxy/zap-extensions/pull/4388/files#diff-917ac965d05701d9e7539db10451980f651e871b90e5c2ef8cc09966d348c219R8

**Comment by decuple-dream** [comment]
@thc202, Thanks to push request!

 I didn't know how to pull for a specific commit, so I modified 'zap-extension/addOns/browserView/browserView.gradle.kts' to the same commit. (line 1) And I tried './gradlew copyZapAddOn' again, and after a while, the following error occurs:
```
./gradlew copyZapAddOn
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :addOns:browserView
Project :addOns:browserView => no module-info.java found

> Task :addOns:oast:enhance
DataNucleus Enhancer completed with success for 1 classes.

> Task :addOns:paramdigger:generateZapAddOnManifest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':addOns:paramdigger:generateZapAddOnManifest'.
```

**Comment by thc202** [comment]
Most likely caused by an OOME, try running `./gradlew -Dorg.gradle.jvmargs=-Xmx2g copyZapAddOn`. (If it went that far it means this Browser View build issue is addressed with that change.)

**Comment by decuple-dream** [comment]
```
./gradlew -Dorg.gradle.jvmargs=-Xmx2g copyZapAddOn

> Configure project :addOns:browserView
Project :addOns:browserView => no module-info.java found

> Task :addOns:oast:enhance
DataNucleus Enhancer completed with success for 1 classes.

BUILD SUCCESSFUL in 1m 6s
3420 actionable tasks: 1641 executed, 1779 up-to-date
```

I think it worked, If 'zap-hud' is built and executed with the same command, an error occurs.
```
./gradlew copyZapAddOn

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nodeSetup'.
> Failed to query the value of task ':nodeSetup' property 'nodeArchiveFile'.
   > Could not resolve all files for configuration ':detachedConfiguration1'.
      > Could not find org.nodejs:node:10.24.1.
        Searched in the following locations:
          - https://repo.maven.apache.org/maven2/org/nodejs/node/10.24.1/node-10.24.1.pom
          - https://nodejs.org/dist/v10.24.1/node-v10.24.1-darwin-arm64.tar.gz
        Required by:
            project :

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

except for `zap-hud`, go to the zaproxy folder and run `./gradlew run`

<img width="525" alt="스크린샷 2023-02-23 오전 8 39 32" src="https://user-images.githubusercontent.com/67303734/220788288-ac82877b-1901-40c2-8574-f14ba780a04b.png">
<img width="828" alt="스크린샷 2023-02-23 오전 8 37 51" src="https://user-images.githubusercontent.com/67303734/220788120-8d6fd7b5-b6a0-426e-b4f1-d21ddd5c822e.png">
<img width="826" alt="스크린샷 2023-02-23 오전 8 38 27" src="https://user-images.githubusercontent.com/67303734/220788150-e9bd60df-192a-4a90-ab71-9369db174681.png">

`zap-extension` is built properly, right? thanks to reply! @thc202 <3

**Comment by decuple-dream** [comment]
Umm, And I did it with Burp(Hand over to Zap upstream from Burp)
However, an error message is displayed in the response.
```
ZAP Error [org.apache.hc.core5.http2.H2StreamResetException]: Stream reset (1)
Stack Trace:
org.apache.hc.core5.http2.H2StreamResetException: Stream reset (1)
	at org.apache.hc.core5.http2.impl.nio.AbstractH2StreamMultiplexer.consumeFrame(AbstractH2StreamMultiplexer.java:871)
	at org.apache.hc.core5.http2.impl.nio.AbstractH2StreamMultiplexer.onInput(AbstractH2StreamMultiplexer.java:445)
	at org.apache.hc.core5.http2.impl.nio.AbstractH2IOEventHandler.inputReady(AbstractH2IOEventHandler.java:65)
	at org.apache.hc.core5.http2.impl.nio.ClientH2IOEventHandler.inputReady(ClientH2IOEventHandler.java:39)
	at org.apache.hc.core5.reactor.ssl.SSLIOSession.decryptData(SSLIOSession.java:600)
	at org.apache.hc.core5.reactor.ssl.SSLIOSession.access$200(SSLIOSession.java:74)
	at org.apache.hc.core5.reactor.ssl.SSLIOSession$1.inputReady(SSLIOSession.java:202)
	at org.apache.hc.core5.reactor.InternalDataChannel.onIOEvent(InternalDataChannel.java:142)
	at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:51)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processEvents(SingleCoreIOReactor.java:178)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:127)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:86)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Unknown Source)
```
It's the same even if you start with the latest release just in case. T^T

**Comment by decuple-dream** [comment]
I closed it by mistake. I'm sorry.

**Comment by thc202** [comment]
https://github.com/zaproxy/zaproxy/issues/7707#issuecomment-1441008261
Did you update the HUD repo? That error shouldn't happen in latest `main` (hopefully).

Yes, `zap-extensions` built correctly.

https://github.com/zaproxy/zaproxy/issues/7707#issuecomment-1441021688
Seems that Burp is not happy with what ZAP sent. I've seen that behaviour when HTTP/2 messages are semantically incorrect (e.g. mixed case headers, invalid or unknown pseudo headers…).

**Comment by decuple-dream** [comment]
I just cloned and built the ZAP-HUD GitHub, but I get the same error.
```
Could not determine the dependencies of task ':nodeSetup'.
> Failed to query the value of task ':nodeSetup' property 'nodeArchiveFile'.
   > Could not resolve all files for configuration ':detachedConfiguration1'.
      > Could not find org.nodejs:node:10.24.1.
        Searched in the following locations:
          - https://repo.maven.apache.org/maven2/org/nodejs/node/10.24.1/node-10.24.1.pom
          - https://nodejs.org/dist/v10.24.1/node-v10.24.1-darwin-arm64.tar.gz
        Required by:
            project :
```
I put the ZAP port from the burp to the Upstream. ZAP is not satisfied with what Burp sent, right?
Thank you so much for your busy schedule.

**Comment by thc202** [comment]
The HUD should now be fixed (thanks to zaproxy/zap-hud#1200).

I thought the setup was ZAP <-> Burp <-> Server ?
